### PR TITLE
pkg/types: small refactor to make code more readable

### PR DESCRIPTION
**Pull request by mqliang**
@xiang90 I am recently reading the source code of ETCD, hoping to get a in-depth understanding of it. I will try to refactor some code to make it better if I meet some place needs to be improved. Are you guys fine with this kind of PR(no big feature or bug-fix, just small refactor)?

### CODE REVIEW

File: pkg/types/urlsmap.go [hunk 1]
```diff
@@ -25,9 +25,12 @@ type URLsMap map[string]URLs
 
 // NewURLsMap returns a URLsMap instantiated from the given string,
 // which consists of discovery-formatted names-to-URLs, like:
-// mach0=http://1.1.1.1:2380,mach0=http://2.2.2.2::2380,mach1=http://3.3.3.3:2380,mach2=http://4.4.4.4:2380
+// mach0=http://1.1.1.1:2380,mach1=http://2.2.2.2::2380,mach2=http://3.3.3.3:2380,mach3=http://4.4.4.4:2380
```
Comment (xiang90) [inline pkg/types/urlsmap.go]: i think the two mach0s is intended. we want to show that we do accept duplicate key but different urls.

Probably a better example could be 
```
mach0=http://1.1.1.1:2380,mach0=http://1.1.1.2::2380,mach1=http://2.2.2.2:2380,mach2=http://3.3.3.3:2380
```

File: pkg/types/urlsmap.go [hunk 2]
```diff
@@ -71,23 +74,19 @@ func (c URLsMap) Len() int {
...
-			continue
-		}
-		value := ""
-		if i := strings.Index(key, "="); i >= 0 {
-			key, value = key[:i], key[i+1:]
+	if s == "" {
+		return m, nil
+	}
+	pairs := strings.Split(s, ",")
+	for _, pair := range pairs {
```
Comment (xiang90) [inline pkg/types/urlsmap.go]: is there a reason we do not want to keep the previous logic? I believe it was copied from go's stdlib somewhere... so i assume it should just work.

Comment (xiang90) [inline pkg/types/urlsmap.go]: ok. i guess your solution is stricter than the previous. so parsing a=b,,c=d will return an error?

Comment (mqliang) [inline pkg/types/urlsmap.go]: > is there a reason we do not want to keep the previous logic

No special reason, I was just thinking using `strings.Split()` was more readable.  

BTW, is there a reason about the implementation of https://github.com/coreos/etcd/blob/master/pkg/fileutil/fileutil.go#L39

Implemented as
```
func IsDirWriteable(dir string) error {
	return syscall.Access(dir, syscall.O_RDWR)
}
```
is much more elegant, right?

Comment (xiang90) [inline pkg/types/urlsmap.go]: not really. when you use syscall pkg, you need to worry about portability (windows, freebsd etc..)

Comment (xiang90) [inline pkg/types/urlsmap.go]: let's change back to the prev impl. and then this pr looks good to me.

Comment (mqliang) [inline pkg/types/urlsmap.go]: Thanks. I am going to close this.

**Pull request closed without merging**